ssl/ssl_lib.cc

```diff
@@ -3047,6 +3047,37 @@ uint16_t SSL_get_peer_signature_algorithm(const SSL *ssl) {
   return session->peer_signature_algorithm;
 }
 
+int SSL_get_peer_signature_nid(const SSL *ssl, int *psig_nid) {
+  uint16_t sig_alg = SSL_get_peer_signature_algorithm(ssl);
+  if (sig_alg == 0) {
+    return 0;
+  }
+
+  const EVP_MD *digest_type = SSL_get_signature_algorithm_digest(sig_alg);
+  if (digest_type == nullptr) {
+    return 0;
+  }
+
+  *psig_nid = EVP_MD_nid(digest_type);
```

Comment:
Is there an error case to check for like `EVP_PKEY_NONE` below?

Reply:
Error case is checked when SSL_get_signature_algorithm_digest is called. So there should always be a valid mapping for any given EVP_MD object (EVP_MD_nid calls EVP_md_type which just returns md->type).